### DISCUSSION TITLE
Like your images very much!
Can you creat one for Roonbridge as well?
Thanks

### DISCUSSION THREAD

**Discussion by KokonZhang** [GioF71/mpd-alsa-docker#280] [Ideas]

**Comment by GioF71** [comment]
Hello! First of all, thank you for your appreciation and kind words.
I have to tell you that I do not currently have a Roon subscription. Maybe I will try a trial period, but I do not feel like spending that kind of money on Roon, even if I know its value, also considering that I just purchased a MinimServer License. I know, it's not the same, but still.

So, consider that I did not reallty try anything I am about to suggest you for the aforementioned reason.
As far as I know, Roon can be used against SqueezeLite endpoints.
See [here](https://antipodes.support/t/step-by-step-guide-to-using-squeezelite-with-roon/575) and [here](https://community.roonlabs.com/t/roon-cant-find-squeezebox-devices/124331). You might already know that I have a repo for squeezelite on docker.
Also, I just read today that there is some upnp bridge for Roon: see [here](https://community.roonlabs.com/t/ifi-bridge-a-upnp-dlna-app-solution-for-playback-on-roon/85864). This *might* work against mpd and upmpdcli (guess it, I have prepared docker images for both).

That said, I have tried building a dedicated docker container for roon-bridge, if you want to try it. You would be the first tester.
See [here](https://github.com/GioF71/roon-bridge-docker) the repo.
You will have to build the image yourself, it really takes a few moments. I tried on amd64 and armhf (a Tinkerboard, so also a Pi3/Pi4 should work). Of course I did not really *try*, but at least the software installs correctly and seems to also run correctly.

Have fun, please report success/failure here or, even better, on that repo.

**Reply by KokonZhang** [reply]
Thank you so much.
Yes I do know the Squeezelite method, but still prefer a direct connection. I will try this new repo and let you know.

**Reply by KokonZhang** [reply]
Currently I am using your mpd-alsa and upmpdcli repo, also have minimserver starter version installed on OMV nas.

**Comment by GioF71** [comment]
Hello, I tried myself, and it worked for me. I had to add a new variable (`FORCE_ARCH`) in order to support a weird setup with LibreElec on the Pi4.
I don't know if I like the Roon experience though.
I hope it worked for you, please share your experience.
I also published docker images. Future support is unlikely considering the cost of the service.

**Comment by KokonZhang** [comment]
Unfortunatly, this repo doesn't work on my Fedora core os on x86-64 platform, even can't finish curl downloading. I am trying https://hub.docker.com/repository/docker/elgeeko/roon-bridge, and face "Warn:  get lock file path: /tmp/.rnbhgem0-"

**Comment by GioF71** [comment]
Hello, that warning about the lock file happens to me as well. About the download error, try without the volume for `/roon`, it might be the same issue as with the mpd-alsa container. 
Here's my output on manjaro x86_64:

```text
➜  roon-bridge git:(master) ✗ docker-compose up

[+] Running 8/8
 ⠿ roon-bridge Pulled                                                                                                                 13.3s
   ⠿ bb73b8dc2908 Pull complete                                                                                                       11.1s
   ⠿ 4f4fb700ef54 Pull complete                                                                                                       11.3s
   ⠿ db4cda3e241e Pull complete                                                                                                       11.3s
   ⠿ b70d4baa6097 Pull complete                                                                                                       11.4s
   ⠿ 16ff0b1a6db1 Pull complete                                                                                                       11.5s
   ⠿ d1420ab45aba Pull complete                                                                                                       11.6s
   ⠿ e3f78ab426b0 Pull complete                                                                                                       11.6s
[+] Running 1/1
 ⠿ Container roon-bridge  Created                                                                                                      0.0s
Attaching to roon-bridge
roon-bridge  | Install bridge
roon-bridge  |   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
roon-bridge  |                                  Dload  Upload   Total   Spent    Left  Speed
100 16.5M  100 16.5M    0     0  7861k      0  0:00:02  0:00:02 --:--:-- 7861k
roon-bridge  | Installation terminated
roon-bridge  | Start bridge
roon-bridge  | 00:00:00.002 Warn:  get lock file path: /tmp/.rnbgem0-
roon-bridge  | 00:00:00.119 Trace: [childprocess] using unix child process
roon-bridge  | 00:00:00.137 Debug: PathForResource, filename: ../.update
roon-bridge  | 00:00:00.143 Debug: PathForResource, candidate: /roon/RoonBridge/Bridge/../.update
roon-bridge  | 00:00:00.145 Debug: PathForResource, filename: RAATServer
roon-bridge  | 00:00:00.145 Debug: PathForResource, candidate: /roon/RoonBridge/Bridge/RAATServer
roon-bridge  | 00:00:00.145 Debug: PathForResource, filename: RoonBridgeHelper
roon-bridge  | 00:00:00.145 Debug: PathForResource, candidate: /roon/RoonBridge/Bridge/RoonBridgeHelper
roon-bridge  | Initializing
roon-bridge  | 00:00:00.158 Info:  Starting /roon/RoonBridge/Bridge/RoonBridgeHelper
roon-bridge  | 00:00:00.166 Info:  ConnectOrStartAndWaitForExit RAATServer, path: /roon/RoonBridge/Bridge/RAATServer
roon-bridge  | Not Running (.o)
roon-bridge  | 00:00:00.008 Warn:  get lock file path: /tmp/.rnbhgem0-
roon-bridge  | Running
^CGracefully stopping... (press Ctrl+C again to force)
Aborting on container exit...
[+] Running 0/1
```

This is the compose file:

```text
---
version: "3"

services:
  roon-bridge:
    container_name: roon-bridge
    image: giof71/roon-bridge:latest
    network_mode: host
    devices:
      - /dev/snd:/dev/snd
    volumes:
      - /var/cache/docker-data/roon-bridge:/roon
    labels:
      - com.centurylinklabs.watchtower.enable=false
    restart: unless-stopped
```

Maybe remove the volumes. It will only cause the container to download the files at every container creation.
Let me know if this changes something. Maybe post the container log so it's easier to understand the issue.

EDIT: typos

**Comment by GioF71** [comment]
I made a new release, the volume "/roon" is not needed. We can now cache binary files using a /files volume or by providing the url to a different server, if needed. However the downloads are very small.

Let me know if this helps

**Comment by KokonZhang** [comment]
Thanks for your effort, it works perfect now.



---- Replied Message ----
| From | Giovanni ***@***.***> |
| Date | 03/29/2023 17:24 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [GioF71/mpd-alsa-docker] Like your images very much!=0ACan you creat one for Roonbridge as well?=0AThanks (Discussion #280) |

I made a new release, the volume "/roon" is not needed. We can now cache binary files using a /files volume or by providing the url to a different server, if needed. However the downloads are very small.

Let me know if this helps

**Comment by GioF71** [comment]
Glad to hear that :-)
Consider starring the repos you use if you want, that would be great for me.
Thanks
